ts/packages/node/src/stores/XmlFile.ts

```diff
@@ -0,0 +1,41 @@
+import xml2js from 'xml2js';
+import { type Config } from '@configu/ts';
+import _ from 'lodash';
+import { FileConfigStore } from './File';
+
+export type XmlFileConfigStoreConfiguration = { path: string };
+
+export class XmlFileConfigStore extends FileConfigStore {
+  constructor({ path }: XmlFileConfigStoreConfiguration) {
+    const initialFileState = `
+<?xml version="1.1" encoding="UTF-8"?>
+<root/>
+    `;
+    super('xml-file', { path, initialFileState });
+  }
+
+  parse(fileContent: string): Config[] {
+    let output: Config[] = [];
+    const parser = new xml2js.Parser({ explicitArray: false });
+    parser.parseString(fileContent, function (error, result) {
+      if (error) {
+        throw error;
+      }
+      output = result.root.config;
+    });
+
+    return output;
+  }
```

Comment:
The new strategy implementation is great, but a bug makes the store unusable when the file does not exist. This is because when the root tag is empty, it returns an empty string, resulting in the parse method giving a wrong response type.

![image](https://github.com/configu/configu/assets/47181182/7a462efa-d589-4843-9ca7-45febd535699)
![image](https://github.com/configu/configu/assets/47181182/577720eb-f97f-4e37-84fa-fbd1cb5f45f1)

To fix this, I suggest the following changes:
- Leverage the `emptyTag` property to return an empty array when no configs are under the root tag.
- if `result.root.config` is `undefined`, use the empty tag value to return an empty array.
- (Unrelated change) convert the callback to an arrow function to align with the project Eslint rules.

```suggestion
  parse(fileContent: string): Config[] {
    let output: Config[] = [];
    const parser = new xml2js.Parser({ explicitArray: false, emptyTag: () => [] });
    parser.parseString(fileContent, (error, result) => {
      if (error) {
        throw error;
      }
      output = result.root.config ?? result.root;
    });

    return output;
  }
```